### ISSUE TITLE
Distillation and fine-tuning on specific dataset

### ISSUE THREAD

**Issue by samuruph** [tianweiy/CausVid#39]
Hi, great work again!
Would be your method useful to distill a model on a specific dataset (e.g., nuScenes) to have output resembling that dataset distribution? Or it would require further fine-tuning of the teacher model ?
Thank you! :)

**Comment by leinyun-1** [comment]
look at my issue, i'm trying to directly lora finetune the distilled model on specific dataset and i get poor results.  In my view, distilling  a modle on a specific dataset would only result in a model that resembles the teacher distribution because of the dmd loss formulation.

**Comment by samuruph** [comment]
I think since the proposed distillation is only using output of the teacher model, it will not work. 
Probably introducing the proposed GAN loss with real data, would get this objective.
Otherwise probably fine-tuning the entire teacher model + DMD could also work.
What do you think?

**Comment by 5aurabhpathak** [comment]
> I think since the proposed distillation is only using output of the teacher model, it will not work. Probably introducing the proposed GAN loss with real data, would get this objective. Otherwise probably fine-tuning the entire teacher model + DMD could also work. What do you think?

I am also trying to use a different video dataset for distillation. **Unfortunately, the generator loss actually increases instead of decreasing even after finetuning.** By finetuning, I mean I am pretraining the generator on the default list of four denoising steps only: [1000, 757, 522, 0] on new data. Then I am plugging the generator in the DMD setup. The rationale behind finetuning only on the 4 denoising steps is that these are the only steps queried by the DMD pipeline in the CausVid implementation. For the real and fake score I am still using the Wan 1.3B as originally used without any finetuning.

There are two other options I can think of that I have not tried yet.
1. Fully finetune on the new data prior to DMD distillation.
2. Try replacing real and/or fake score models with the 4 step finetuned model in the DMD pipeline.

Do you think this approach makes sense?
